projects/components/src/datagrid/datagrid.component.ts

```diff
@@ -436,7 +453,7 @@ export class DatagridComponent<R extends B, B = any> implements OnInit, AfterVie
         if (this.datagrid.selection.current && this.datagrid.selection.current.length) {
             return this.datagrid.selection.current;
         }
-        return [];
+        return EMPTY_ARRAY;
```

Comment:
This will solve the issue when there is no selection. However, in the case of single election grids, the problem you mentioned above in your comment on line 44 would still persist.
I think a better(not the best) soln would be to assign the Datagrid selection to a variable and assign that to the selected entities of the action menu.

```
private _datagridSelection = [];
@Input()
    get datagridSelection(): B[] {
        if (this.datagrid.selection.currentSingle) {
            this._datagridSelection.length = 0;
            this._datagridSelection.push(this.datagrid.selection.currentSingle);
        }
        if (this.datagrid.selection.current && this.datagrid.selection.current.length) {
            this._datagridSelection = this.datagrid.selection.current;
        }
        this._datagridSelection.length = 0;
    }